changes/2055-layday.md

```diff
@@ -0,0 +1 @@
+add two overload variants to `validate_arguments` for the nested decorator signature
```

Comment:
Maybe something more user oriented like "improve/fix typing of `validate_arguments` to be compliant with `mypy`"

Reply:
Changed to:

> fix annotation of `validate_arguments` when passing configuration as argument